test/schema_test.cc

```diff
@@ -490,3 +503,474 @@ TEST(SchemaTest, NestedDuplicateFieldIdError) {
   EXPECT_THAT(result.error().message,
               ::testing::HasSubstr("Duplicate field id found: 1"));
 }
+
+namespace {
+
+namespace TestFields {
+iceberg::SchemaField Id() { return {1, "id", iceberg::int32(), true}; }
+iceberg::SchemaField Name() { return {2, "name", iceberg::string(), false}; }
+iceberg::SchemaField Age() { return {3, "age", iceberg::int32(), true}; }
+iceberg::SchemaField Email() { return {4, "email", iceberg::string(), true}; }
+iceberg::SchemaField Street() { return {11, "street", iceberg::string(), true}; }
+iceberg::SchemaField City() { return {12, "city", iceberg::string(), true}; }
+iceberg::SchemaField Zip() { return {13, "zip", iceberg::int32(), true}; }
+iceberg::SchemaField Theme() { return {24, "theme", iceberg::string(), true}; }
+iceberg::SchemaField Key() { return {31, "key", iceberg::int32(), false}; }
+iceberg::SchemaField Value() { return {32, "value", iceberg::string(), false}; }
+iceberg::SchemaField Element() { return {41, "element", iceberg::string(), false}; }
+
+}  // namespace TestFields
+
+struct TestSchemaFactory {
+  static std::shared_ptr<iceberg::Schema> BasicSchema(int32_t schema_id = 100) {
```

Comment:
```suggestion
  static std::shared_ptr<iceberg::Schema> BasicSchema() {
```

`schema_id` is not used and unnecessary. Same for below.